packages/lesswrong/server/migrations/2022-08-29-syncEmailsFields.ts

```diff
@@ -0,0 +1,58 @@
+import { registerMigration } from './migrationUtils';
+import Users from '../../lib/collections/users/collection';
+import fs from 'fs';
+import Papa from 'papaparse';
+
+registerMigration({
+  name: "syncEmailsFields",
+  dateWritten: "2022-08-29",
+  idempotent: true,
+  action: async () => {
+    const totalUsersMissingEmail =  await Users.find({email: null}).fetch()
+    const totalRecentUsersMissingEmail =  await Users.find({email: null, lastNotificationsCheck: {$exists:true}}).fetch()
+
+    // eslint-disable-next-line no-console
+    console.log(`There are ${totalUsersMissingEmail.length} users missing an email`)
+    // eslint-disable-next-line no-console
+    console.log(`There are ${totalRecentUsersMissingEmail.length} users missing an email who've checked notifications in past 3 years`)
+
+    const usersEmailsToEmail = await Users.find({email: null, emails:{$ne:null}}).fetch()
+    const usersGoogleToEmail = await Users.find({email: null, 'services.google.email':{$ne:null}}).fetch()
+    const usersGithubToEmail = await Users.find({email: null, 'services.github.email':{$ne:null}}).fetch()
+    const usersFacebookToEmail = await Users.find({email: null, 'services.facebook.email':{$ne:null}}).fetch()
```

Comment:
(Similarly)[https://github.com/ForumMagnum/ForumMagnum/pull/5572/files#r964244689], I think we should be able to derive this in code?